### ISSUE TITLE
Catalog access gives 500 error

### ISSUE THREAD

**Issue by ghost** [SkeltonThatcher/aws-terraform-rancher-single-node-ha#1]
I'm experiencing an issue which I think might be egress related . 

I can't access the catalog , getting a "fail" error page. 
When I try to create a service I get 

"Failed to find network for networkMode managed" 

when trying to pull an image.

May be related to issues with ALB use ? 

"We no longer recommend Application Load Balancer (ALB) in AWS over using the Elastic/Classic Load Balancer (ELB)"

[link](https://docs.rancher.com/rancher/v1.5/en/installing-rancher/installing-server)

**Comment by richbos** [comment]
Hi Paul, yes, this is somewhat perplexing indeed. Thank you for raising. We had already an issue open with rancher themselves and have sent the server container logs from a failed build - https://github.com/rancher/rancher/issues/8372

Many thanks for highlighting the ALB advisory, that could quite possibly be it, and it is interesting as they (rancher) advised switching to ALB only a few months back. As such we updated our plan (from ELB to ALB), however we shall regress and test this week to see if it resolves things. If we have success I shall update the plan and notify accordingly.

**Comment by richbos** [comment]
To update this issue. We have carried out extensive testing today with https/ssl enabled ELB deployments and received the same error. As such we can confirm the issue is not related to ALB use.

**Comment by ghost** [comment]
Thanks for the update, I also modified the deployment to ELB , and got the
same results.
Please keep me updated , I appreciate you guys looking at this.
I'm hoping to use this platform demo as part of a business case

**Comment by richbos** [comment]
Hi Paul, to update again, we have tested today with Rancher v1.4.2 (server) and catalog access is OK, no problems at all.

So, we can verify that we only get the 500 error when using Rancher v1.51, v1.5.2 or v1.5.3. With ELB or ALB, RDS, and either RancherOS or Ubuntu.

I have updated the issue with rancher themselves.

**Comment by richbos** [comment]
We can now verify that this is an issue when using RDS (or a remote DB), in our case using RancherOS and Rancher v1.5.0/5.1/5.2/5.3.

I have updated the main rancher repo issue with our findings.

https://github.com/rancher/rancher/issues/8372

**Comment by richbos** [comment]
Hi, I can confirm that we have now resolved this issue.

The `aws_db_instance.rancher` var from `server.tf` which feeds the `userdata_srv.template` was using the `endpoint` attribute. This was doubling up the port for dial tcp and adding an extra 3306 (i.e. 3306:3306).

Changing the attribute to use `address` (as follows) has resolved the catalog access issue.

`database_address  = "${aws_db_instance.rancher.address}"`

We have also fixed the Rancher versions to v1.5.3 (server) and v1.2.1 (agent).
